cmd/entrypoint/runner_test.go

```diff
@@ -183,10 +185,41 @@ func TestRealRunnerTimeout(t *testing.T) {
 	defer cancel()
 
 	if err := rr.Run(ctx, "sleep", "0.01"); err != nil {
-		if !errors.Is(err, context.DeadlineExceeded) {
+		if !errors.Is(err, entrypoint.ErrContextDeadlineExceeded) {
 			t.Fatalf("unexpected error received: %v", err)
 		}
 	} else {
 		t.Fatalf("step didn't timeout")
 	}
 }
+
+func TestRealRunnerCanceled_beforeCmdWait(t *testing.T) {
+	rr := realRunner{}
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+
+	if err := rr.Run(ctx, "sleep", "3"); err != nil {
+		if !errors.Is(err, entrypoint.ErrContextCanceled) {
+			t.Fatalf("unexpected error received: %v", err)
+		}
+	} else {
+		t.Fatalf("step didn't cancel")
+	}
+}
+
+func TestRealRunnerCanceled(t *testing.T) {
+	rr := realRunner{}
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		time.Sleep(1 * time.Second)
```

Comment:
nit: would it make sense to run `rr.Run` in a goroutine instead and cancel the context? This would avoid the time delay before the test succeeds and may be less likely to be flaky.

Reply:
removed this test

Reply:
I think this is probably a useful test; I'd rather have it around without the suggestions I made than have it removed!

Reply:
thanks @lbernick , I merged the two tests into `TestRealRunnerCancel`, and canceled at a random time, trying to test all the situations